### PR TITLE
Added proper 65sc02, 65ce02 and 4510 detection.

### PR DESCRIPTION
**Pull request by blackystardust**
Added proper 65sc02, 65ce02 and 4510 detection.

### CODE REVIEW

File: libsrc/common/getcpu.s [hunk 1]
```diff
@@ -27,20 +29,39 @@ _getcpu:
...
+        lda     #3              ; CPU_4510 constant
+        nop
+        bne     @L9
+
+; check for 65sc02
+
+@L6:    ldy     $f7
+        ldx     #$00
+        stx     $f7
+        .byte   $f7,$f7         ; nop nop on 65sc02, smb7 $f7 on 65c02 and 65816
```
Comment (greg-king5) [inline libsrc/common/getcpu.s]: According to the documents that I have, the 65816 CPU is an enhanced 16-bit version of the 65sc02, not the 65c02.  It doesn't have the `BBRn`, `BBSn`, `RMBn`, and `SMBn` instructions.

`$f7` is `"sbc [dp],y"` on that CPU.

Comment (blackystardust) [reply libsrc/common/getcpu.s]: You're right, I'll need to fix that check, thank you for pointing it out.

**Comment by blackystardust** [comment]
Fixed the 65816/65802 detection issue.

**Comment by greg-king5** [comment]
:+1:

**Comment by oliverschmidt** [comment]
As far as I can tell this is ready to be merged...

**Comment by oliverschmidt** [comment]
Maybe http://cc65.github.io/doc/funcref.html#ss3.108 "wants" to be adjusted...

**Pull request merged**